src/tigerbeetle/benchmark_load.zig

```diff
@@ -260,343 +272,238 @@ const Generator = union(enum) {
     print_batch_timings: bool,
-    id_order: cli.Command.Benchmark.IdOrder,
-    batch_account_ids: std.ArrayListUnmanaged(u128),
-    batch_transfer_ids: std.ArrayListUnmanaged(u128),
-
-    fn create_account(b: *Benchmark) tb.Account {
-        const random = b.rng.random();
-
-        defer b.account_index += 1;
-        return .{
-            .id = b.account_id_permutation.encode(b.account_index + 1),
-            .user_data_128 = random.int(u128),
-            .user_data_64 = random.int(u64),
-            .user_data_32 = random.int(u32),
-            .reserved = 0,
-            .ledger = 2,
-            .code = 1,
-            .flags = .{
-                .history = b.flag_history,
-                .imported = b.flag_imported,
-            },
-            .debits_pending = 0,
-            .debits_posted = 0,
-            .credits_pending = 0,
-            .credits_posted = 0,
-            .timestamp = if (b.flag_imported) b.account_index + 1 else 0,
-        };
-    }
 
-    fn create_accounts(b: *Benchmark) void {
-        if (b.account_index >= b.account_count) {
-            b.create_transfers();
-            return;
-        }
-
-        // Reset batch.
-        b.batch_accounts.clearRetainingCapacity();
+    // State:
+    clients_busy: std.StaticBitSet(constants.clients_max) =
+        std.StaticBitSet(constants.clients_max).initEmpty(),
+    clients_request_ns: [constants.clients_max]u64 = .{undefined} ** constants.clients_max,
+    client_requests: []align(constants.sector_size) [constants.message_body_size_max]u8,
+    client_replies: []align(constants.sector_size) [constants.message_body_size_max]u8,
+    request_latency_histogram: []u64,
+    request_index: usize = 0,
+    account_index: usize = 0,
+    transfer_index: usize = 0,
+    transfers_created: usize = 0,
+    query_index: usize = 0,
+    stage: Stage = .idle,
+
+    const Stage = enum {
+        idle,
+        register,
+        create_accounts,
+        create_transfers,
+        get_account_transfers,
+        validate_accounts,
+        validate_transfers,
+    };
 
-        // Fill batch.
-        while (b.account_index < b.account_count and
-            b.batch_accounts.items.len < b.batch_accounts.capacity)
-        {
-            b.batch_accounts.appendAssumeCapacity(b.create_account());
+    pub fn run(b: *Benchmark, stage: Stage) void {
+        assert(b.stage == .idle);
+        assert(b.clients.len > 0);
+        assert(b.clients_busy.count() == 0);
+        assert(stdx.zeroed(std.mem.sliceAsBytes(b.request_latency_histogram)));
+        assert(b.request_index == 0);
+        assert(b.account_index == 0);
+        assert(b.transfer_index == 0);
+        assert(b.query_index == 0);
+        assert(stage != .idle);
+
+        b.stage = stage;
+        b.timer.reset();
+
+        for (0..b.clients.len) |_| {
+            switch (b.stage) {
+                .register => b.register(),
+                .create_accounts => b.create_accounts(),
```

Comment:
I wonder if the code would be simpler to understand if the client index is passed as an argument into these function, instead of each of them using `.request_client` for finding the next free client. I _think_ we won't even need the "find first unset bit" logic at all? We are statically assigning work to clients. 

What is confusing is that each `b.create_accounts()` is itself a loop (the callback calls `b.create_accounts()` recursively), and it feels like each iteration of this inner loop might pick a different client, but I think that''s not actually possible. 

But the code is fine either way!